pydm/main_window.py

```diff
@@ -478,3 +480,14 @@ def _confirm_quit(self, callback):
                 QMessageBox.Yes | QMessageBox.No)
             if quit_message == QMessageBox.Yes:
                 callback()
+
+    def contextMenuEvent(self, event):
+        menu = QMenu()
+        show_macro_action = menu.addAction("Show Macros...")
```

Comment:
Hmm, how about the tool bar menu? I think this should be linked from there primarily (see references to `self.ui.menuTools` elsewhere in the file)

@jbellister-slac - would setting `contextMenuEvent` here get in the way of the custom context menu / tool handler that PyDM has?

Reply:
Yes, I know plots in particular get affected by this, might be other widgets too. I took a quick check of this and it will clobber the right click menu on any PyDM plot doing it this way. (For example try running examples/timeplot/timeplot.ui and right clicking on it).

I think the tool bar menu would be the easiest place to put it, would take some extra thinking/work to get it to work with a context menu like this.